tools/amend-event-data.js

```diff
@@ -183,15 +183,36 @@ const patches = {
       matched: 7,
       change: { interface: 'DragEvent', bubbles: true }
     },
+    // The "HTMLElement" base interface receives most HTML events in theory,
+    // but some of the events only fire on specific HTML elements in practice.
+    // The following updates refine the target interfaces of these events.
+    // (This is not a temporary fix: "HTMLElement" is the correct target
+    // interface from a spec perspective, that's where the event handlers are
+    // defined)
+    // Also, the "cancel" event bubbles on input elements but not on other
+    // target interfaces, so we need to duplicate the entry in the extract.
     {
       pattern: { type: "cancel"},
       matched: 1,
-      change: { targets: ["HTMLDialogElement", "HTMLInputElement"] }
+      change: { targets: ["HTMLInputElement"] }
+    },
+    {
+      add: {
+        type: "cancel",
+        interface: "Event",
+        bubbles: false,
+        targets: ["CloseWatcher", "HTMLDialogElement"],
+        href: "https://html.spec.whatwg.org/multipage/indices.html#event-cancel",
+        src: {
+          format: "summary table",
+          href: "https://html.spec.whatwg.org/multipage/indices.html#event-cancel"
+        }
+      }
     },
     {
       pattern: { type: "close"},
       matched: 1,
-      change: { targets: ["HTMLDialogElement"   ] }
+      change: { targets: ["CloseWatcher", "HTMLDialogElement", "MessagePort"] }
```

Comment:
is this meant to be part of that patch?

Reply:
Forgot to mention it, but yes. Not needed to solve `cancel`, but needed to refine the target interfaces of `close`.

Reply:
1. can you add a comment to that effect?
2. shouldn't MessagePort and CloseWatcher be fixable upstream? or is it that the aggregation can't deal properly with this case?

Reply:
> can you add a comment to that effect?

Where? Or about what? I added a generic comment in the code already a few lines above that one precisely to explain why we have all of these patches for HTML events: to have something more specific than `HTMLElement`, which is the interface that HTML references because it defines the event handler.

> shouldn't MessagePort and CloseWatcher be fixable upstream? or is it that the aggregation can't deal properly with this case?

That has already been fixed upstream with the PR you sent. That's not why this patch exists. As for the other HTML events, the patch is meant to transform `HTMLElement`, here into `HTMLDialogElement`. I'm adding `CloseWatcher` and `MessagePort` because our patching mechanism is "all or nothing". If I don't repeat `CloseWatcher` and `MessagePort`, it effectively removes them, unfixing the fix that was made upstream...

Reply:
Thanks for the explanation and sorry for the noise 😊